### PR TITLE
Northwind SQL Queries

### PR DESCRIPTION
**Pull request by spore9**
Issue #131 
All queries in HomeWork4.sql stored procedure.

### CODE REVIEW

File: HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+/*1.1.2*/
+	SELECT OrderID, CASE WHEN ShippedDate IS NULL THEN 'Not shipped' END AS ShippedDate FROM Orders WHERE ShippedDate IS NULL;
+/*1.1.3*/
+	SELECT OrderID AS 'Order Number', 
+	CASE WHEN ShippedDate IS NULL 
+	THEN 'Not shipped' 
+	ELSE CONVERT(varchar,ShippedDate,101) 
+	END AS 'Shipped Date' 
+	FROM Orders 
+	WHERE ShippedDate IS NULL OR ShippedDate>='5/6/1998';
```
Comment (MilRoad) [inline HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql]: In the task we need to choose orders after this date, so there must be ">", not ">=".

Comment (spore9) [reply HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql]: Agree, my mistake

File: HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql [hunk 2]
```diff
@@ -0,0 +1,115 @@
+﻿CREATE PROCEDURE [dbo].[HomeWork4]
+AS
+/*1.1.1*/
+	SELECT OrderID, ShippedDate, ShipVia FROM Orders WHERE ShipVia >= 2 AND ShippedDate>='5/6/1998';
```
Comment (Foxtra) [inline HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql]: May be it is only my local problem, but this query returns only 3 rows. 
I would rewrite it to `ShippedDate >= (1998-05-06)`

File: HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+/*1.1.2*/
+	SELECT OrderID, CASE WHEN ShippedDate IS NULL THEN 'Not shipped' END AS ShippedDate FROM Orders WHERE ShippedDate IS NULL;
+/*1.1.3*/
+	SELECT OrderID AS 'Order Number', 
+	CASE WHEN ShippedDate IS NULL 
+	THEN 'Not shipped' 
+	ELSE CONVERT(varchar,ShippedDate,101) 
+	END AS 'Shipped Date' 
+	FROM Orders 
+	WHERE ShippedDate IS NULL OR ShippedDate>'5/6/1998';
```
Comment (Foxtra) [inline HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql]: Again the same issue, I see only 'Not shipped' orders,
`ShippedDate>(1998-05-06)`

File: HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	FROM Products  
+	WHERE ProductName LIKE '%cho_olade%';
+/*2.1.1*/
+	SELECT SUM(UnitPrice*Quantity*(1-Discount)) as 'Totals' 
+	FROM [Order Details]  
+/*2.1.2*/
+	SELECT COUNT(OrderDate) - COUNT(ShippedDate) as 'Not shipped'
+	FROM Orders;
+/*2.1.3*/
+	SELECT COUNT(DISTINCT CustomerID)
```
Comment (Foxtra) [inline HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql]: You can add here some name of column, for example "Amount of customers"

File: HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	FROM Customers
+	LEFT JOIN Orders ON Orders.CustomerID = Customers.CustomerID
+	GROUP BY Customers.CustomerID, Customers.ContactName
+	ORDER BY COUNT(OrderID)
+/*2.4.1*/
+	SELECT CompanyName
+	FROM Suppliers
+	WHERE 0 IN (SELECT UnitsInStock  FROM Products WHERE Suppliers.SupplierID = Products.SupplierID)
+/*2.4.2*/
+	SELECT CONCAT(FirstName, ' ', LastName)
```
Comment (Foxtra) [inline HW4/Kirill_Uryvaev/NorthwindQueries/NorthwindQueries/dbo/Stored Procedures/131-HomeWork4.sql]: and name of the column here for example 'Employee'

**Pull request merged**